dvc/path_info.py

```diff
@@ -120,11 +120,17 @@ def __repr__(self):
 class URLInfo(_BasePath):
     DEFAULT_PORTS = {"http": 80, "https": 443, "ssh": 22, "hdfs": 0}
 
-    def __init__(self, url):
+    def __init__(self, url, keep_host_case=False):
         p = urlparse(url)
         assert not p.query and not p.params and not p.fragment
         assert p.password is None
-        self._fill_parts(p.scheme, p.hostname, p.username, p.port, p.path)
+
+        if keep_host_case:
+            host = p.netloc.split(":")[0]
+        else:
+            host = p.hostname
+
+        self._fill_parts(p.scheme, host, p.username, p.port, p.path)
 
```

Comment:
Do I understand that this is not needed, as we've been restoring it GDriveURLInfo before?

Reply:
Before adding here, I had removed from `GDriveURLInfo`. I did this change because the previous version wasn't working correctly during a debugging scenario. Yes, I know, it doesn't make sense, but it's the truth.

Reply:
@julsal So is this actually needed for this PR? If not, let's not do it.

Reply:
Done!